src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryCachingConfiguration.java

```diff
@@ -97,8 +97,17 @@ public FormValidation doClearCache(@QueryParameter String name) throws Interrupt
                         if (libraryNamePath.readToString().equals(name)) {
                             FilePath libraryCachePath = LibraryCachingConfiguration.getGlobalLibrariesCacheDir()
                                     .child(libraryNamePath.getName().replace("-name.txt", ""));
-                            libraryCachePath.deleteRecursive();
-                            libraryNamePath.delete();
+                            ReentrantReadWriteLock retrieveLock = LibraryAdder.getReadWriteLockFor(libraryCachePath.getName());
+                            if (retrieveLock.writeLock().tryLock(10, TimeUnit.SECONDS)) {
```

Comment:
I think there should be a "force cache clear" option that ignores locks because doClearCache is an on demand action. This preserves the existing behavior and offers a solution other than filesystem manipulation if a retrieve.lock is stuck. This can be a new checkbox on the UI.